Toolkit/AppDeployToolkit/AppDeployToolkitMain.ps1

```diff
@@ -15362,29 +15362,25 @@ https://psappdeploytoolkit.com
     }
     Process {
         Try {
-            $ServiceObject = Get-WmiObject -ComputerName $ComputerName -Class 'Win32_Service' -Filter "Name='$Name'" -ErrorAction 'Stop'
-            # If nothing is returned from Win32_Service, check Win32_BaseService
-            If (-not $ServiceObject) {
-                $ServiceObject = Get-WmiObject -ComputerName $ComputerName -Class 'Win32_BaseService' -Filter "Name='$Name'" -ErrorAction 'Stop'
+            If ($PassThru) {
+                $ServiceObject = Get-WmiObject -ComputerName $ComputerName -Class 'Win32_Service' -Filter "Name='$Name'" -ErrorAction 'Stop'
+                # If nothing is returned from Win32_Service, check Win32_BaseService
+                If (-not $ServiceObject) {
+                    $ServiceObject = Get-WmiObject -ComputerName $ComputerName -Class 'Win32_BaseService' -Filter "Name='$Name'" -ErrorAction 'Stop'
+                }
+            }
+            else {
+                # Known issue that WMI method fails in Sandbox, so only use it if -PassThru requested
+                $ServiceObject = (Get-Service -ComputerName $ComputerName -Name $Name -ErrorAction Ignore).Count -gt 0
```

Comment:
This can't use `-ErrorAction Ignore` is 3.x needs to remain compatible with PowerShell 2.x targets.

Reply:
TIL, I'll change that! It's used in a couple of other places that I resolved also.